yt/utilities/lib/pixelization_routines.pyx

```diff
@@ -1910,6 +2095,61 @@ def rotate_particle_coord(np.float64_t[:] px,
     return px_rotated, py_rotated, rot_bounds_x0, rot_bounds_x1, rot_bounds_y0, rot_bounds_y1
 
 
+@cython.boundscheck(False)
```

Comment:
So I think this function looks right.  What I'm wondering is if it needs to be in Cython as-is.  I think it could potentially take advantage of being in cython by consolidating the `matmul` stuff with the copying of coordinates -- i.e., iterating over the particles one-by-one and avoiding temporary arrays.  I'm not 100% sure that it's *necessary*, but it could potentially eliminate some allocations.

Reply:
ya, I agree -- it could be written with just numpy matrix operations at the python level or rewritten with a different loop control. 

Also, could actually cut out an entire loop over the particles by calculating the rotated particle positions as needed within `pixelize_sph_kernel_projection_with_depth_check` (and `pixelize_sph_kernel_projection`)?